src/decision/src/subroutines/bottom_line/AdjustToLine.cpp

```diff
@@ -0,0 +1,44 @@
+//
+// Created by drdc-s2632 on 17/07/18.
+//
+
+#include "AdjustToLine.h"
+
+std::vector<ros::Subscriber>
+AdjustToLine::getSubscriptions(ros::NodeHandle nh) {
+    std::string gateDetectTopic = "line_detect_output";
+
+    std::vector<ros::Subscriber> subs;
+    subs.push_back(
+    nh.subscribe(gateDetectTopic, 10, &AdjustToLine::lineDetectCallback, this));
+    return subs;
+}
+
+void AdjustToLine::lineDetectCallback(
+const line_detect::LineDetectMsg::ConstPtr& msg) {
+    geometry_msgs::TwistStamped command;
+
+    if (msg->angleToParallelFrontMarker >
+        (*constants_)["ERROR_TOLERANCE_LINE_ANGLE"]) {
+        command.twist.angular.z = TWISTRIGHT;
+        publishCommand(command);
+        return;
+    } else if (msg->angleToParallelFrontMarker <
+               -(*constants_)["ERROR_TOLERANCE_LINE_ANGLE"]) {
+        command.twist.angular.z = TWISTLEFT;
+        publishCommand(command);
+        return;
+    }
+
+    if (msg->lateralDistanceFromFrontMarker >
+        (*constants_)["ERROR_TOLERANCE_LINE_LATERAL_DISTANCE"]) {
+        command.twist.linear.y = LEFT;
+        publishCommand(command);
```

Comment:
what's the rationale for only allowing one of `angular.z` and `linear.y` to be sent each pass through?

Reply:
The lateral distance measurement is dependent on us being perpendicular to the line, so first we have to adjust to be perpendicular then the lateral distance will be more accurate. Just kind of a hack to make sure errors aren't accumulated